arraysandpointers.tex

```diff
@@ -210,63 +191,67 @@ \subsection{Multidimensional Dynamic Array}
 }
 \end{codeblock}
 
-Now you should know everything you need to allocate an array dynamically.
-Also note that you can always allocate an array with size 1 if you only need 1 element!
-In that case it is usually more common to use the \texttt{*} operator for dereferencing rather than \texttt{[]}.
+Since we are allocating multiple separate pieces of memory here, it is not enough to simply free \lstinline!arr!.
+We must be careful to free every element of \lstinline!arr! (the rows) before we free \lstinline!arr!.
+
+As with the nested arrays we saw earlier in the chapter, there is nothing special about our choice of \lstinline!arr! as an array of rows rather than an array of columns.
+
+The other commonly used method for dynamically-sized multidimensional arrays is to allocate a single block of memory large enough to hold every element, and calculate the indexes required for specific elements.
```

Comment:
Perhaps an explanation on why you'd use this over the conceptually easier method